tests.contrib.aiohttp.test_request_safety.TestAiohttpSafety testMethod=test_multiple_full_request randomly fails

```
_________________ TestAiohttpSafety.test_multiple_full_request _________________

self = <tests.contrib.aiohttp.test_request_safety.TestAiohttpSafety testMethod=test_multiple_full_request>

    @unittest_run_loop
    @asyncio.coroutine
    def test_multiple_full_request(self):
        # it should produce a wrong trace, but the Context must
        # be finished
        def make_requests():
            url = self.client.make_url('/delayed/')
            response = request.urlopen(str(url)).read().decode('utf-8')
            eq_('Done', response)
    
        # blocking call executed in different threads
        ctx = self.tracer.get_call_context()
        threads = [threading.Thread(target=make_requests) for _ in range(10)]
        for t in threads:
            t.daemon = True
            t.start()
    ˘
        # we should yield so that this loop can handle
        # threads' requests
        yield from asyncio.sleep(0.5)
        for t in threads:
            t.join(timeout=0.5)
    
        # the trace is wrong but the Context is finished
        traces = self.tracer.writer.pop_traces()
>       eq_(1, len(traces))
E       AssertionError: 1 != 2
```

See https://circleci.com/gh/DataDog/dd-trace-py/110366

--- a/tests/contrib/aiohttp/test_request_safety.py
+++ b/tests/contrib/aiohttp/test_request_safety.py
@@ -68,14 +68,13 @@ class TestAiohttpSafety(TraceTestCase):
         ctx = self.tracer.get_call_context()
         threads = [threading.Thread(target=make_requests) for _ in range(10)]
         for t in threads:
-            t.daemon = True
             t.start()
 
-        # we should yield so that this loop can handle
-        # threads' requests
-        yield from asyncio.sleep(0.5)
         for t in threads:
-            t.join(timeout=0.5)
+            # we should yield so that this loop can handle
+            # threads' requests
+            yield from asyncio.sleep(0.1)
+            t.join(0.1)
 
         # the trace is wrong but the Context is finished
         traces = self.tracer.writer.pop_traces()

